web/new_alt_text_manager.js

```diff
@@ -0,0 +1,396 @@
+        if (this.#isAILoading) {
+          this.#addAltText(altText);
+        }
+      }
+    } catch (e) {
+      console.error(e);
+      hasError = true;
+    }
+
+    this.#toggleLoading(false);
+
+    if (hasError && this.#uiManager) {
+      this.#toggleError(true);
+      this.#toggleTitle();
+      this.#toggleDisclaimer();
+    }
+  }
+
+  #addAltText(altText) {
+    if (!this.#uiManager || this.#textarea.value) {
+      return;
+    }
+    this.#textarea.value = altText;
+  }
+
+  async editAltText(uiManager, editor, firstTime) {
+    if (this.#currentEditor || !editor) {
+      return;
+    }
+
+    if (firstTime && editor.hasAltTextData()) {
+      editor.altTextFinish();
+      return;
+    }
+
+    this.#firstTime = firstTime;
+    let { mlManager } = uiManager;
+    if (!mlManager?.isReady("altText")) {
+      mlManager = null;
+    }
+    const isAltTextEnabledPromise = mlManager?.isEnabledFor("altText");
+
+    this.#currentEditor = editor;
+    this.#uiManager = uiManager;
+    this.#uiManager.removeEditListeners();
+
+    ({ altText: this.#previousAltText } = editor.altTextData);
+    this.#textarea.value = this.#previousAltText ?? "";
+
+    // TODO: get this value from Firefox
+    //   (https://bugzilla.mozilla.org/show_bug.cgi?id=1908184)
+    const AI_MAX_IMAGE_DIMENSION = 224;
+
+    // The max dimension of the preview in the dialog is 180px, so we keep 224px
+    // and rescale it thanks to css.
+
+    let canvas;
+    if (mlManager) {
+      ({ canvas, imageData: this.#imageData } = editor.copyCanvas(
+        AI_MAX_IMAGE_DIMENSION,
+        /* createImageData = */ true
+      ));
+      this.#toggleGuessAltText(
+        await isAltTextEnabledPromise,
+        /* isInitial = */ true
+      );
+    } else {
+      ({ canvas } = editor.copyCanvas(
+        AI_MAX_IMAGE_DIMENSION,
+        /* createImageData = */ false
+      ));
+    }
+
+    canvas.setAttribute("role", "presentation");
+    this.#imagePreview.append(canvas);
+
+    this.#toggleNotNow();
+    this.#toggleAI(!!mlManager);
+    this.#toggleError(false);
+
+    try {
+      await this.#overlayManager.open(this.#dialog);
+    } catch (ex) {
+      this.#close();
+      throw ex;
+    }
+  }
+
+  #cancel() {
+    this.#currentEditor.altTextData = {
+      cancel: true,
+    };
+    this.#finish();
+  }
+
+  #finish() {
+    if (this.#overlayManager.active === this.#dialog) {
+      this.#overlayManager.close(this.#dialog);
+    }
+  }
+
+  #close() {
+    const canvas = this.#imagePreview.firstChild;
+    canvas.remove();
+    canvas.width = canvas.height = 0;
+    this.#imageData = null;
+
+    this.#currentEditor._reportTelemetry(
+      this.#telemetryData || {
+        action: "alt_text_cancel",
+      }
+    );
+
+    this.#telemetryData = null;
+    this.#toggleLoading(false);
+
+    this.#uiManager?.addEditListeners();
+    this.#currentEditor.altTextFinish();
+    this.#uiManager.setSelected(this.#currentEditor);
```

Comment:
Shouldn't we use `this.#uiManager?.setSelected` here given the statement two lines above?